api/dp-openapi.yaml

```diff
@@ -502,6 +502,113 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/codepoints:
+    get:
+      tags:
+      - project
+      description: Returns a string representing valid codepoints for the project
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Codepoints
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/codepoints'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/pages/checkout:
+    post:
+      tags:
+      - project
+      description: Checks out a page to the user
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: roundid
+        in: query
```

Comment:
The roundid here is a query, but in the path below for `/projects/{projectid}/pages/{pagename}/rounds/{roundid}`, do we want them consistent?

Reply:
See some of the other discussions on the PR. The gist is that we don't want them consistent because while a project can't be in more than one round we want the ability to say "check out a page and the project should be in round X" and if the project isn't in round X then return an error.